ui/components/input/base/_index.scss

```diff
@@ -330,4 +330,11 @@ ul.slds-dropdown-list {
   width: 100%;
 }
 
+.slds-large-size--0-of-12, 
+.slds-medium--0-of-12, 
+.slds-medium-size--0-of-12, 
+.slds-size--0-of-12 {
```

Comment:
I think these sizes need to be in the format

.slds-large-size_0-of-12,
.slds-medium-size_0-of-12,
.slds-size_0-of-12

Do we need the slds-medium--0-of-12? shouldn't it always be slds-medium-size_0-of-12?

Reply:
I tried to keep this as default way. Another default generated classes for 12 grid are
      .slds-small-size--1-of-12 
      .slds-medium-size--1-of-12
      .slds-large-size--1-of-12
      .slds-small-size--2-of-12

So I think, should be same. Let me know if you still want in that format.

Reply:
I'm having second thoughts about even adding these.

I'm assuming these are used when someone sets a Element to be 0 width. If that's the case, we should hide it completely (which is what the `display: none` does), but in addition we should avoid rendering unnecessary DOM elements, so if we used an `ng-hide` instead to handle this then we don't even render the DOM elements. This would have the added benefit of giving a performance improvement because we'll have fewer watchers and fewer DOM nodes.

What do you think @avnesh-neev?